app/lib/email.server.ts

```diff
@@ -89,6 +91,35 @@ function maybeThrowSES(e: any, warning: string) {
   ])
 }
 
+/**
+ * Amazon SES has a byte limit of 262144 for TemplateData arguments.
+ * If the circular body is over this size limit, the message will be truncated.
+ * If the body is truncated, the circular website link will be removed.
+ * A warning that the email has been truncated is added and the circular website link is added back in.
+ * An additional buffer is allowed for for variation in url length and truncation warning.
+ */
+export function truncateEmailBodyIfNecessary({
+  body,
+  circularId,
+}: {
+  body: string
+  circularId?: number
```

Comment:
In what situation would `circularId` be undefined?

Reply:
I'm not sure what you are asking. when this issue was assigned to me, you said to look at this comment that you made: https://github.com/nasa-gcn/gcn.nasa.gov/issues/2899#issuecomment-2798899590 and to do that instead of just make a size check. that way the circular in the database would be complete and the email would be truncated to prevent issue.